aspnetapp/README.md

```diff
@@ -1,36 +1,95 @@
 
 ```console
 git clone https://github.com/dotnet/dotnet-docker-samples/
-cd dotnet-docker-samples/aspnetapp
 ```
 
-Follow these steps to run the sample locally:
+## Build and run the sample locally
+
+You can build and run the sample locally with the [.NET Core 2.0 SDK](https://www.microsoft.com/net/download/core) using the following instructions. The instructions assume that you are in the root of the repository.
 
 ```console
-dotnet restore
-dotnet run
+cd aspnetapp
+dotnet run -c release
 ```
 
-Follow these steps to run this sample in a Linux or Windows container:
+After the application starts, visit `http://localhost:8000` in your web browser.
+
+Note: The `-c release` argument builds the application in release mode (the default is debug mode). See the [dotnet run reference](https://docs.microsoft.com/dotnet/core/tools/dotnet-run) for more information on commandline parameters.
+
+## Build and run the sample with Docker for Linux containers
+
+You can build and run the sample in Docker using Linux containers with the following commands. The instructions assume that you are in the root of the repository.
 
 ```console
+cd aspnetapp
 docker build -t aspnetapp .
-docker run -d -p 80:80 aspnetapp
+docker run -it --rm -p 8000:80 aspnetapp
 ```
 
-## View your web page running from your container
-* If you are using a linux container you can simply browse to http://localhost:80 to access your app in a web browser.
-* If you are using the Nano [Windows Container](https://docs.docker.com/docker-for-windows/) there is currently a bug that affects how [Windows 10 talks to Containers via "NAT"](https://github.com/Microsoft/Virtualization-Documentation/issues/181#issuecomment-252671828) (Network Address Translation). Today you must hit the IP of the container directly. You can get the IP address of your container with the following steps:
-  1. Run `docker ps` and copy the hash of your container ID
-  3. Run `docker inspect -f "{{ .NetworkSettings.Networks.nat.IPAddress }}" HASH` where `HASH` is replaced with your container ID.
-  4. Copy the container ip address and paste into your browser with port 80. (Ex: 172.16.240.197:80)
\ No newline at end of file
+After the application starts, visit `http://localhost:8000` in your web browser.
+
+Note: The `-p` argument maps port 8000 on you local machine to port 80 in the container (the form of the port mapping is `host:container`). See the [Docker run reference](https://docs.docker.com/engine/reference/commandline/run/) for more information on commandline paramaters.
+
+## Build and run the sample with Docker for Windows containers
+
+You can build and run the sample in Docker using Windows containers with the following commands. The instructions assume that you are in the root of the repository.
+
+```console
+cd aspnetapp
+docker build -t aspnetapp .
+docker run -d --rm aspnetapp
```

Comment:
Consider naming the container and making use of it in the `docker ps` step below.  Naming service containers such as this is a good practice IMO.